### PR TITLE
Feature: home sections

### PR DESCRIPTION
**Pull request by edbrett**
Time to start making some pages:
- Section component to handle home page repeated sections that accepts background image and sets css grid
- Children inside each section comp with custom elements and components
- Add themes to button
- New background image and icons 😮 
- New entries in settings.scss
- basic video frame using react-player

NOTES: waiting for pages PR with new component folder structure. Will update when ready.

UPDATE: now merged with latest pages folder structure, and all page components moved to match.

### CODE REVIEW

File: app/javascript/app/components/nav/nav-component.jsx [hunk 1]
```diff
@@ -90,7 +91,8 @@ class NavBar extends PureComponent {
 
 NavBar.propTypes = {
   countriesOpen: Proptypes.bool,
-  setCountriesVisibility: Proptypes.func.isRequired
+  setCountriesVisibility: Proptypes.func.isRequired,
+  location: Proptypes.object
```
Comment (j8seangel) [inline app/javascript/app/components/nav/nav-component.jsx]: I'd add this as required.

File: app/javascript/app/components/nav/nav.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
 import actions from './nav-actions';
 
 export { default as reducers } from './nav-reducers';
 export { default as actions } from './nav-actions';
 
-const mapStateToProps = state => ({
-  countriesOpen: state.nav.countriesOpen
+const mapStateToProps = (state, { location }) => ({
+  countriesOpen: state.nav.countriesOpen,
+  location
```
Comment (j8seangel) [inline app/javascript/app/components/nav/nav.js]: We don't need this as we are wrapping with the `withRouter` HOC

File: app/javascript/app/components/section/section-component.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import styles from './section-styles.scss';
+
+const Section = ({ children, backgroundImage, className }) =>
+  (<section
+    className={styles.section}
+    style={{
+      backgroundImage: `url(${backgroundImage})`
+    }}
+  >
+    <div className={cx(className, layout.content, styles.doubleFold)}>
```
Comment (j8seangel) [inline app/javascript/app/components/section/section-component.jsx]: To be able to override the default styles I'd move the className to the end of the list.

File: package.json [hunk 4]
```diff
@@ -24,6 +24,7 @@
     "react-css-themr": "^2.1.2",
     "react-dom": "^15.6.1",
     "react-motion": "^0.5.0",
+    "react-player": "^0.20.0",
```
Comment (j8seangel) [inline package.json]: Please remember to install the exact version so we'll avoid problems with package-lock.json later.

File: app/javascript/app/pages/home/home-component.jsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import homeThreeBg from 'assets/backgrounds/home_bg_3';
+import homeThreeImage from 'assets/backgrounds/home_image_3';
+
+import styles from './home-styles.scss';
+
+const Home = () =>
+  (<div>
+    <Section className={styles.homeOne} backgroundImage={homeOneBg}>
+      <div className={styles.column}>
+        <Icon icon={cwLogo} className={styles.cwLogo} theme={styles} />
```
Comment (j8seangel) [inline app/javascript/app/pages/home/home-component.jsx]: Why do we need to have all the styles of the homePage in the theme prop to the icon component? I also think they are not being used.

File: app/javascript/app/pages/home/home-component.jsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        />
+      </div>
+    </Section>
+    <Section className={styles.homeTwo} backgroundImage={homeTwoBg}>
+      <div className={styles.column}>
+        <img src={homeTwoImage} alt="home-section-two" />
+      </div>
+      <div className={styles.column}>
+        <Intro
+          className={styles.lightIntro}
```
Comment (j8seangel) [inline app/javascript/app/pages/home/home-component.jsx]: Sounds weird for me to have classNames with styles and theme using the same styles object, let's discuss it later!

File: app/javascript/app/components/section/section-styles.scss [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+.doubleFold {
+  @extend %grid;
+
+  grid-template-columns: repeat(2, 1fr);
+}
+
+.section {
+  display: block;
+  background-repeat: none;
```
Comment (j8seangel) [inline app/javascript/app/components/section/section-styles.scss]: That prop doesn't exists, please use `no-repeat`

File: app/javascript/app/components/nav/nav.js (hunk 2 above)
Comment (edbrett) [inline app/javascript/app/components/nav/nav.js]: Nice.

File: package.json (hunk 4 above)
Comment (edbrett) [inline package.json]: Ok.

File: app/javascript/app/components/nav/nav-component.jsx (hunk 1 above)
Comment (edbrett) [inline app/javascript/app/components/nav/nav-component.jsx]: Ok.

**Pull request merged**